smartfile/__init__.py

```diff
@@ -124,8 +129,36 @@ def put(self, endpoint, id=None, **kwargs):
     def post(self, endpoint, id=None, **kwargs):
         return self._request('post', endpoint, id=id, data=kwargs)
 
-    def delete(self, endpoint, id=None, **kwargs):
-        return self._request('delete', endpoint, id=id, data=kwargs)
+    def delete(self, deletefile):
+        return self.post('/path/oper/remove', path=deletefile)
+
+    def upload(self, filename, fileobj):
+        if filename.endswith('/'):
+            raise ValueError("File name should have no trailing slash")
+        arg = (filename, fileobj)
+        return self.post('/path/data/', file=arg)
+
+    def download(self, file_to_be_downloaded):
+        """ file_to_be_downloaded is a file-like object that has already
+        been uploaded, you cannot download folders """
+        # download uses shutil.copyfileobj to download, which copies
+        # the data in chunks
+        o = open(file_to_be_downloaded, 'wb')
+        return shutil.copyfileobj(self.get('/path/data/',
+                                  file_to_be_downloaded), o)
+
+    def move(self, sourcefile, destination):
+        # check destination folder for / at end
+        if not destination.endswith("/"):
+            destination = destination + "/"
+        # check destination folder for / at begining
+        if not destination.startswith("/"):
+            destination = "/" + destination
+        t = self.post('/path/oper/move/', src=sourcefile, dst=destination)
+        while True:
+            s = self.get('/task', t['uuid'])
+            if s['result']['status'] == 'SUCCESS':
+                break
```

Comment:
Mmmm this looks pretty stressful for the server. I would add a `time.sleep(0.2)` at the very least.
The ideal way to do this would be via a websocket but that's another story.
Also `if s['result']['status'] == 'SUCCESS':` looks incorrect because if the action is never successful (e.g. there's an error) you will end up looping forever.